homeassistant/components/velbus/__init__.py

```diff
@@ -50,31 +51,85 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data[DOMAIN][entry.entry_id] = controller
     await controller.connect()
 
+    # add the controller as a device
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device_registry.async_get_or_create(
+        name=f"{entry.title}-interface",
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, entry.data[CONF_PORT])},
+        connections={(CONF_PORT, entry.data[CONF_PORT])},
```

Comment:
To register a device in the device registry we need to have an identifier or connection identifier that is both unique and stable. The port config option is not stable, right? Ie, it could change for the same device if the user re-configures the config entry.

Can we get a serial number, mac address or similar? Otherwise maybe it's a better option to have the service take the port as a service data parameter and we don't register a device.

Reply:
as the port/interface string is the only available identifier, and this string can be a serial port, a ip:port combination this is difficult. 
So i switched the services to the port string